frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx

```diff
@@ -29,6 +29,7 @@ const hiddenBackgroundStyle = css`
 `;
 
 export const DashCardRoot = styled.div<DashCardRootProps>`
+  overflow: hidden;
```

Comment:
This breaks dashboard card actions. That's why a few E2E tests have failed.
![image](https://github.com/metabase/metabase/assets/1937582/3397a0b1-8726-4a83-8225-fce95b60a192)

There are a few solutions I think we could try to overcome this. But all solutions including removing `overflow: hidden` here.

1. We could use the node that Embed Frame theme is applying the background to.
    https://github.com/metabase/metabase/blob/4982cc70e09a9f258e296ba4f47e6cb27d7c3f74/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css#L41-L44
    
    Then we only need to modify the selector where the image is downloaded here https://github.com/metabase/metabase/blob/4982cc70e09a9f258e296ba4f47e6cb27d7c3f74/frontend/src/metabase/query_builder/actions/downloading.ts#L48-L49
    Currently we don't have a static selector for that node, but we could maybe convert `data-testid` to `data-element-id` which we used for selectors that are used outside tests. e.g. https://github.com/metabase/metabase/blob/4982cc70e09a9f258e296ba4f47e6cb27d7c3f74/frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx#L453
    then we could use the `:has` selector like this `[data-element-id=dashcard]:has(<old selector>)`. I tested that this solution works fine.
2. Just round the `[data-card-key]` border when not exporting to image. But I don't like that we set background to many components. Now we set to dashcard, and in this PR we also repeat the background for the visualization card again.

Reply:
I have this code on my feature branch that should be easy to 
change and support this use case too https://github.com/metabase/metabase/pull/43743/files#diff-f23559c04621d6e009e0fc27fc865262397f3c02504b08d0b412ad531962ce27R15-R21

Something like
```diff
diff --git a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css b/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css
index eb73ef1e27..f55f586246 100644
--- a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css
+++ b/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css
@@ -15,7 +15,7 @@ body {
 .ThemeNight.EmbedFrame,
 /* this is to make it work when exporting to pdfs,
 where the EmbedFrame is not part of the exported dom */
-.ThemeNight.EmbedFrame > .WithThemeBackground {
+.ThemeNight.EmbedFrame .WithThemeBackground {
   background-color: var(--mb-color-bg-black);
   border-color: var(--mb-color-bg-dark);
 }
diff --git a/frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx b/frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx
index b9cb99cd46..0996bcd689 100644
--- a/frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx
+++ b/frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx
@@ -14,6 +14,7 @@ import DashboardS from "metabase/css/dashboard.module.css";
 import * as MetabaseAnalytics from "metabase/lib/analytics";
 import { formatNumber } from "metabase/lib/formatting";
 import { equals } from "metabase/lib/utils";
+import EmbedFrameS from "metabase/public/components/EmbedFrame/EmbedFrame.module.css";
 import { getIsShowingRawTable } from "metabase/query_builder/selectors";
 import { getIsEmbeddingSdk } from "metabase/selectors/embed";
 import { getFont } from "metabase/styled-components/selectors";
@@ -518,7 +519,12 @@ class Visualization extends PureComponent {
           ) : (
             <div
               data-card-key={getCardKey(series[0].card?.id)}
-              className={cx(CS.flex, CS.flexColumn, CS.flexFull)}
+              className={cx(
+                CS.flex,
+                CS.flexColumn,
+                CS.flexFull,
+                EmbedFrameS.WithThemeBackground,
+              )}
             >
               <CardVisualization
                 {...this.props}
```

I'll try and extract the needed logic on its own PR

Reply:
@npretto Would it be possible for your feature to not have 2 backgrounds? https://github.com/metabase/metabase/blob/aedc0162e83c90b2b3d731dbd1164e6526bb6df0/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css#L15-L18

It's the exact same scenario we're facing here. I'm worried this would complicate theming in the SDK in the future.

We could modify the selector when exporting to PDF to target `EmbedFrame` couldn't we?

https://github.com/metabase/metabase/blob/aedc0162e83c90b2b3d731dbd1164e6526bb6df0/frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx#L289

Reply:
This would only have any effect with .ThemeNight, which shouldn't be used with the sdk with a custom theme i think.

EmbedFrame also has other things we don't want to export, it could be possible to select that, but then we have to hide a few things

Reply:
That's fair, can you add a comment about that then since it's not obvious why we couldn't target `embedframe` directly.

Reply:
Extracted the code in this PR: https://github.com/metabase/metabase/pull/45023
I also reproed it on the release branch of v50 so I added the backport label to the PR

Reply:
Hey @WiNloSt and @npretto, thanks for your feedback and ideas. I changed the approach and it now should tick all the boxes — https://github.com/metabase/metabase/pull/44941/commits/97b64c65cf1c3cab7c0f051d5ee7a15fc830bb93

It also should fix the bug when we had a dashboard with N same cards connected to different filters which lead to showing different data on them. Previously the png export selector used CardId which is wrong and it should've been DashcardId

Reply:
@npretto good catch, thanks! I added `important` which seems ok here since we're overriding it only for a moment while exporting a png